src/mon/OSDMonitor.cc

```diff
@@ -1364,6 +1365,21 @@ bool OSDMonitor::should_propose(double& delay)
     return true;
   }
 
+  // propose as fast as possible if updating up_thru or pg_temp
+  // want to merge OSDMap changes as much as possible
+  if ((pending_inc.new_primary_temp.size() == 1
+      || pending_inc.new_up_thru.size() == 1)
+      && pending_inc.new_state.size() < 2) {
+    dout(15) << " propose as fast as possible for up_thru/pg_temp" << dendl;
+
+    utime_t now = ceph_clock_now();
+    if (now - last_attempped_minwait_time > g_conf->paxos_propose_interval) {
```

Comment:
This needs to look at the last map generated without using minwait as well -- it might be more recent!